### PR TITLE
Improved URL handling, changed default browser and variable to disable default mappings

### PR DESCRIPTION
**Pull request by Jorenar**
* Some URL (e.g. `http://www.egr.unlv.edu/~ed/assembly64.pdf#page=21&zoom=auto,-304,543`) would be opened partially (`http://www.egr.unlv.edu/~ed/assembly64.pdf#page=21`)
* `xdg-open` is currently most commonly used for opening various files with user default applications
* Added variable allowing to disable default mappings completely
* In README was no mention about available variables

### CODE REVIEW

File: autoload/open_url.vim [hunk 1]
```diff
@@ -5,7 +5,7 @@ function! open_url#open(url)
   elseif has('mac') || has('macunix') || has('gui_macvim')
     exe 'silent !open "'.url.'"'
   else
-    exe 'silent !'.g:open_url_browser_default url
+    exe 'silent! !'.g:open_url_browser_default.' "'.url.'" &> /dev/null &'
```
Comment (dhruvasagar) [inline autoload/open_url.vim:8]: I am not sure if `$> /dev/null` is required, neither is the trailing `&` to send this to background.

File: plugin/open_url.vim [hunk 2]
```diff
@@ -3,13 +3,13 @@ if exists('g:loaded_open_url')
...
-function! s:SetGlobalOpt(opt, val)
-  if !exists('g:'.a:opt)
-    let g:{a:opt} = a:val
-  endif
-endfunction
+if !exists('g:open_url_browser_default')
+  let g:open_url_browser_default = 'xdg-open'
+endif
 
-call s:SetGlobalOpt('open_url_browser_default', 'sensible-browser')
```
Comment (dhruvasagar) [inline plugin/open_url.vim:12]: Why did you remove this ? There's no need to change the default offered by the plugin, if you want to override / change this config, set it in your `~/.vimrc`

File: plugin/open_url.vim [hunk 3]
```diff
@@ -3,13 +3,13 @@ if exists('g:loaded_open_url')
...
+if !exists('g:open_url_default_mappings')
+  let g:open_url_default_mappings = 1
+endif
```
Comment (dhruvasagar) [inline plugin/open_url.vim]: Utilize the helper `call s:SetGlobalOpt('open_url_default_mappings', 1)` for better readability.

File: plugin/open_url.vim [hunk 4]
```diff
@@ -3,13 +3,13 @@ if exists('g:loaded_open_url')
 endif
 let g:loaded_open_url = 1
 
-function! s:SetGlobalOpt(opt, val)
-  if !exists('g:'.a:opt)
-    let g:{a:opt} = a:val
-  endif
-endfunction
```
Comment (dhruvasagar) [inline plugin/open_url.vim]: This is a nice helper, it may not seen useful but results in much cleaner plugin configuration setup which is both easier to read and update. Kindly do not remove this.

**Review by dhruvasagar — CHANGES_REQUESTED** [review]
Thanks for your contribution.

File: autoload/open_url.vim (hunk 1 above)
Comment (Jorenar) [reply autoload/open_url.vim:8]: Those two are necessary, when plugin is opening the first instance of browser (I use `firefox`, I don't know how it is with other browsers, but compatibility with one of the most popular isn't a bad thing, right? ;) )

Without `&` I won't be able to go back to Vim and continue editing while having browser open.

Without `&> /dev/null` buffers get polluted when closing browser with such messages:
```
###!!! [Parent][DispatchAsyncMessage] Error: PLayerTransaction::Msg_ReleaseLayer Processing error: message was deserialized, but the handler returned false (indicating failure)          
                                                                                                                                                                                                                                      
                                                                                                                                                                                                                            [Parent 21844, Gecko_IOThread] WARNING: pipe error (42): Connection reset by peer: file /build/firefox/src/firefox-68.0.1/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 358                                                                  
                                                                                                                                                                            [Parent 21844, Gecko_IOThread] WARNING: pipe error (98): Connection reset by peer: file /build/firefox/src/firefox-68.0.1/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 358                                                                                                                          
                                                                                                                                                                                                                                             
                                                                                                                     ###!!! [Parent][RunMessage] Error: Channel closing: too late to send/recv, messages will be lost                          
```
and there is a need for another `redraw!` but I think this approach is easier than detecting if browser was closed.

File: plugin/open_url.vim (hunk 2 above)
Comment (Jorenar) [reply plugin/open_url.vim:12]: `xdg-open` (`xdg-utils`) is usually installed by default, while `sensible-browser` (`sensible-utils`) not. And even after installing this package, it would still not detect `firefox` and require me to set BROWSER environment variable.
And thus, I changed the default browser, as it should be something what most users already use.

Also, in README was no mention about `g:open_url_browser_default`, so nobody would know that this option exist without reading a code (what - let's be honest - most people don't do).

File: plugin/open_url.vim (hunk 3 above)
Comment (Jorenar) [reply plugin/open_url.vim]: Done

File: plugin/open_url.vim (hunk 4 above)
Comment (Jorenar) [reply plugin/open_url.vim]: OK, this is a nice helper. 

> it may not seen useful

So let's make it useful ;)
